File: admin-backend/application-api/src/main/java/com/xfrog/platform/application/permission/api/RoleApi.java
package com.xfrog.platform.application.permission.api;

import com.xfrog.platform.application.permission.api.dto.CreateRoleRequestDTO;
import com.xfrog.platform.application.permission.api.dto.PermissionItemDTO;
import com.xfrog.platform.application.permission.api.dto.RoleDTO;
import com.xfrog.platform.application.permission.api.dto.UpdateRoleRequestDTO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.List;

@Tag(name = "RoleApi", description = "角色管理接口")
@RequestMapping("/api/roles")
public interface RoleApi {
    @PostMapping
    @Operation(summary = "创建角色")
    Long createRole(@Valid @RequestBody CreateRoleRequestDTO createRoleRequestDTO);

    @GetMapping("/list")
    @Operation(summary = "查询角色列表")
    List<RoleDTO> listRoles();

    @GetMapping("/{roleId}/permission-items")
    @Operation(summary = "获取角色权限列表")
    List<PermissionItemDTO> getRolePermissionItems(@PathVariable("roleId") Long roleId);

    @PostMapping("/{roleId}")
    @Operation(summary = "更新角色")
    void updateRole(@PathVariable("roleId") Long roleId, @Valid @RequestBody UpdateRoleRequestDTO updateRoleRequestDTO);

    @DeleteMapping("/{roleId}")
    @Operation(summary = "删除角色")
    void deleteRole(@PathVariable("roleId") Long roleId);

    @PostMapping("/{roleId}/{enabled}")
    @Operation(summary = "启用或禁用角色")
    void enableRole(@PathVariable("roleId") Long roleId, @PathVariable("enabled") Boolean enabled);

    @PutMapping("/grant-permissions/{roleId}")
    @Operation(summary = "为角色授予权限")
    void grantPermissionItems(@PathVariable(name = "roleId") Long roleId, @RequestBody List<Long> permissionItemIds);
}

File: admin-ui/src/services/swagger/roleApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request';

/** 创建角色 POST /api/roles */
export async function createRole(body: API.CreateRoleRequestDTO, options?: { [key: string]: any }) {
  return request<number>('/api/roles', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 更新角色 POST /api/roles/${param0} */
export async function updateRole(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateRoleParams,
  body: API.UpdateRoleRequestDTO,
  options?: { [key: string]: any },
) {
  const { roleId: param0, ...queryParams } = params;
  return request<any>(`/api/roles/${param0}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 删除角色 DELETE /api/roles/${param0} */
export async function deleteRole(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.deleteRoleParams,
  options?: { [key: string]: any },
) {
  const { roleId: param0, ...queryParams } = params;
  return request<any>(`/api/roles/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 启用或禁用角色 POST /api/roles/${param0}/${param1} */
export async function enableRole(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.enableRoleParams,
  options?: { [key: string]: any },
) {
  const { roleId: param0, enabled: param1, ...queryParams } = params;
  return request<any>(`/api/roles/${param0}/${param1}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 获取角色权限列表 GET /api/roles/${param0}/permission-items */
export async function getRolePermissionItems(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getRolePermissionItemsParams,
  options?: { [key: string]: any },
) {
  const { roleId: param0, ...queryParams } = params;
  return request<API.PermissionItemDTO[]>(`/api/roles/${param0}/permission-items`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 为角色授予权限 PUT /api/roles/grant-permissions/${param0} */
export async function grantPermissionItems(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.grantPermissionItemsParams,
  body: number[],
  options?: { [key: string]: any },
) {
  const { roleId: param0, ...queryParams } = params;
  return request<any>(`/api/roles/grant-permissions/${param0}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 查询角色列表 GET /api/roles/list */
export async function listRoles(options?: { [key: string]: any }) {
  return request<API.RoleDTO[]>('/api/roles/list', {
    method: 'GET',
    ...(options || {}),
  });
}
